diff --git a/Foundation/Components/Sheet/Sheet.lua b/Foundation/Components/Sheet/Sheet.lua
index 95d05c4..5aaa8a9 100644
--- a/Foundation/Components/Sheet/Sheet.lua
+++ b/Foundation/Components/Sheet/Sheet.lua
@@ -1,17 +1,12 @@
 local Foundation = script:FindFirstAncestor("Foundation")
 local Packages = Foundation.Parent
 local React = require(Packages.React)
-local ReactUtils = require(Packages.ReactUtils)
-
-local Flags = require(Foundation.Utility.Flags)
 
 local BottomSheet = require(script.Parent.BottomSheet)
 local CenterSheet = require(script.Parent.CenterSheet)
+local DialogSize = require(Foundation.Enums.DialogSize)
 local SideSheet = require(script.Parent.SideSheet)
 local useOverlay = require(Foundation.Providers.Overlay.useOverlay)
-local useScreen = require(Foundation.Providers.Overlay.useScreen)
-local GuiService = require(Foundation.Utility.Wrappers).Services.GuiService
-local DialogSize = require(Foundation.Enums.DialogSize)
 local withDefaults = require(Foundation.Utility.withDefaults)
 type DialogSize = DialogSize.DialogSize
 
@@ -39,11 +34,11 @@ local function Sheet(sheetProps: SheetProps, ref: React.Ref<GuiObject>): React.R
 		end
 
 		local function update()
-			setSmallDisplay(screen.AbsoluteSize.X <= 500)
+			setSmallDisplay(screen.AbsoluteSize.X <= 640)
 		end
 		update()
 
-		local connection = GuiService:GetPropertyChangedSignal("ViewportDisplaySize"):Connect(update)
+		local connection = screen:GetPropertyChangedSignal("AbsoluteSize"):Connect(update)
 		return function()
 			connection:Disconnect()
 		end
