diff --git a/Foundation/Components/Popover/Content/PopoverContent.lua.rej b/Foundation/Components/Popover/Content/PopoverContent.lua.rej
deleted file mode 100644
index cde91c8..0000000
--- a/Foundation/Components/Popover/Content/PopoverContent.lua.rej
+++ /dev/null
@@ -1,9 +0,0 @@
-diff a/plugin/Packages/_Index/Foundation/Foundation/Components/Popover/Content/PopoverContent.lua b/plugin/Packages/_Index/Foundation/Foundation/Components/Popover/Content/PopoverContent.lua	(rejected hunks)
-@@ -205,6 +205,7 @@ local function PopoverContent(contentProps: PopoverContentProps, forwardedRef: R
- 					Position = position:map(function(value: Vector2)
- 						return UDim2.fromOffset(value.X, value.Y)
- 					end),
-+					InputSink = Enum.InputSink.All,
- 					selection = props.selection,
- 					selectionGroup = props.selectionGroup,
- 					sizeConstraint = {
